nixos/modules/services/home-automation/zwave-js.nix

```diff
@@ -0,0 +1,157 @@
+{config, pkgs, lib, ...}:
+
+with lib;
+
+let
+  cfg = config.services.zwave-js;
+  cacheDir = "/var/cache/zwave-js";
+  runtimeDir = "/run/zwave-js";
+  defaultConfig = {
+    storage = {
+      inherit cacheDir;
+    };
+  };
+  mergedConfigFile = "${runtimeDir}/config.json";
+in {
+
+  options.services.zwave-js = {
+    enable = mkEnableOption (mdDoc "Start the zwave-js server on boot");
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.zwave-js-server;
+      defaultText = literalExpression "pkgs.zwave-js-server";
+      description = mdDoc ''
+        Package for zwave-js server
+      '';
+    };
+
+    port = mkOption {
+      type = types.int;
+      default = 3000;
+      description = mdDoc ''
+        Port for the server to listen on.
+      '';
+    };
+
+    mockDriver = mkOption {
+      type = types.bool;
+      default = false;
+      description = mdDoc ''
+        Use a mock USB serial driver (mutually exclusive with serialPort option).
+      '';
+    };
```

Comment:
Wondering if such an option is even useful outside of our test. If not we could also live with an `extraArgs` option, correct?

Reply:
Definitely debatable, I went with this approach since it seemed a bit weird to require a serialPort in the case the mock driver is being used, and pass it on the command line even though we don't want it to be used.

Reply:
Maybe make it invisible, internal or whatever it is called.

Reply:
Done!